ci-operator/step-registry/cucushift/installer/check/reboot-nodes/cucushift-installer-check-reboot-nodes-commands.sh

```diff
@@ -4,10 +4,376 @@ set -o nounset
+}
+
+# Function to log with timestamp
+log() {
+    echo "$(get_timestamp) - $*"
+}
+
+# Enhanced error handling function
+error_handler() {
+    local exit_code=$?
+    local line_number=$1
+    local command="$2"
+    
+    log "==========================================" >&2
+    log "ERROR: Script failed unexpectedly!" >&2
+    log "==========================================" >&2
+    log "Exit code: ${exit_code}" >&2
+    log "Line number: ${line_number}" >&2
+    log "Failed command: ${command}" >&2
+    log "Current working directory: $(pwd)" >&2
+    log "Current user: $(whoami)" >&2
+    log "Process ID: $$" >&2
+    log "Parent process ID: $PPID" >&2
+    log "Shell options: $-" >&2
+    log "==========================================" >&2
+    log "Stack trace:" >&2
+    local frame=0
+    while caller $frame; do
+        ((frame++))
+    done >&2
+    log "==========================================" >&2
+    log "Environment variables:" >&2
+    env | grep -E "(CLUSTER|AWS|KUBE|OPENSHIFT)" | sort >&2
+    log "==========================================" >&2
+    log "Recent command history:" >&2
+    history | tail -10 >&2
+    log "==========================================" >&2
+    exit 1
+}
+
+# Add ERR trap to catch any unexpected exits
+trap 'error_handler $LINENO "$BASH_COMMAND"' ERR
+
+# Enhanced exit handler
+exit_handler() {
+    local exit_code=$?
+    log "=========================================="
+    log "Script execution completed"
+    log "=========================================="
+    log "Final exit code: ${exit_code}"
+    log "Execution time: $((SECONDS)) seconds"
+    log "Current time: $(get_timestamp)"
+    log "=========================================="
+}
+
+# Add EXIT trap to log script completion
+trap 'exit_handler' EXIT
+
+# Configuration constants - simplified for AWS hard reboot
+readonly AWS_INSTANCE_TIMEOUT=60    # 60 attempts × 10 seconds = 10 minutes
+readonly CLUSTER_RECOVERY_TIMEOUT=120  # 120 attempts × 30 seconds = 60 minutes
+
+# Reboot type configuration
+# HARD_REBOOT: Use cloud provider commands (AWS EC2 stop/start, etc.)
+# SOFT_REBOOT: Use SSH-based reboot (existing behavior)
+readonly REBOOT_TYPE="${REBOOT_TYPE:-SOFT_REBOOT}"
+
+# Record script start time for execution time calculation
+SECONDS=0
+
+log "=========================================="
+log "Script execution started"
+log "=========================================="
+log "Script: $0"
+log "Arguments: $*"
+log "Process ID: $$"
+log "Working directory: $(pwd)"
+log "User: $(whoami)"
+log "=========================================="
+
+# Platform detection and setup using case statement
+case "${CLUSTER_TYPE}" in
+    aws)
+        # Configure AWS credentials - support both CI and local environments
+        export AWS_REGION="${LEASED_RESOURCE}"
+        
+        # Determine AWS credentials file location
+        aws_cred_file=""
+        if [[ -n "${CLUSTER_PROFILE_DIR:-}" && -f "${CLUSTER_PROFILE_DIR}/.awscred" ]]; then
```

Comment:
`${CLUSTER_PROFILE_DIR}/.awscred` is the only location where the AWS credential file is stored, so we don't need to check it.